### PR TITLE
[sBTC DR] Bitcoin client retry logic and prevention of race conditions for Stacks contract calls

### PR DESCRIPTION
**Pull request by stjepangolemac**
- bitcoin client will now retry on known recoverable errors
- we're not keeping track of transaction creation (before it's broadcasted)

### CODE REVIEW

File: romeo/src/state.rs
Comment (netrome) [inline romeo/src/state.rs:73]: Perhaps we should consider adding retry logic here as well. We should not crash on rejected transactions.

**Review by netrome — APPROVED** [review]
So far so good.

**Review by friedger — APPROVED** [review]
LGTM

**Comment by stjepangolemac** [comment]
Had to resolve some conflicts and ask you again for reviews.

File: romeo/src/bitcoin_client.rs
Comment (friedger) [inline romeo/src/bitcoin_client.rs:118]: This could come from a config file for testing

**Review by friedger — DISMISSED** [review]
LGTM

File: romeo/src/state.rs
Comment (jferrant) [reply romeo/src/state.rs:96]: At this point, retry has failed with a permanent error. I assume one reason for this is if the stacks node it is broadcasting to is just down. At this point, do we expect Romeo to log an error and retry? What other things would cause a permanent error? In those cases, would we expect Romeo to skip the transaction?

**Review by jferrant — DISMISSED** [review]

**Review by friedger — APPROVED** [review]
LGTM

**Comment by friedger** [comment]
@jferrant Could please review this again, as I merged main into it. Sorry.

**Review by jferrant — APPROVED** [review]

**Comment by AshtonStephens** [comment]
It looks like it's failing the format check - @stjepangolemac before you merge could you make it meet the format requirements?

**Comment by friedger** [comment]
Seeing the difficulty to get 2 reviews, we have to choose between speed of development and passed checks.

**Comment by stjepangolemac** [comment]
Yeah @AshtonStephens it'll get fixed automatically in one of the upcoming PRs. We'd need at least a few more days to get necessary approvals if it was fixed now.

**Comment by AshtonStephens** [comment]
So this is essentially resolved then?

**Pull request merged**
